yt_dlp/extractor/bilibili.py

```diff
@@ -68,6 +69,7 @@ def extract_formats(self, play_info):
             'tbr': float_or_none(video.get('bandwidth'), scale=1000),
             'filesize': int_or_none(video.get('size')),
             'quality': int_or_none(video.get('id')),
+            'format_id': self._search_regex(r'-(\d+)\.m4s\?', traverse_obj(video, 'baseUrl', 'base_url'), 'video:format_id', fatal=False) or str_or_none(video.get('id')),
```

Comment:
Line is too long, and it's fatal if the traversal returns `None`

IMO you could do it like this:
```suggestion
            'format_id': traverse_obj(
                video, (('baseUrl', 'base_url'), {self._FORMAT_ID_RE.search}, 1),
                ('id', {str_or_none}), get_all=False),
```
and define this above in the base class:
```py
    _FORMAT_ID_RE = re.compile(r'-(\d+)\.m4s\?')
```
and make sure to `import re`

Reply:
Correct me if I am wrong but I thought `_search_regex` would handle `string=None` with `fatal=False`
https://github.com/yt-dlp/yt-dlp/blob/66cc64ff6696f9921ff112a278542f8d999ffea4/yt_dlp/extractor/common.py#L1233-L1241


Still, this change is much easier to read and thanks for the comment.

Reply:
ah you're right, my bad